test/e2e/specs/site-editor/template-revert.spec.js

```diff
@@ -0,0 +1,272 @@
+	},
+} );
+
+test.describe( 'Template Revert', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+	test.beforeEach( async ( { admin, requestUtils } ) => {
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await admin.visitSiteEditor();
+	} );
+
+	test( 'should delete the template after saving the reverted template', async ( {
+		page,
+		templateRevertUtils,
+	} ) => {
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		await page.click( 'role=button[name="Show template details"]' );
+
+		// The revert button isn't visible anymore.
+		await expect(
+			page.locator( 'role=menuitem[name=/Clear customizations/i]' )
+		).not.toBeVisible();
+	} );
+
+	test( 'should show the original content after revert', async ( {
+		templateRevertUtils,
+	} ) => {
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		const contentAfter = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfter );
+	} );
+
+	test( 'should show the original content after revert and page reload', async ( {
+		admin,
+		templateRevertUtils,
+	} ) => {
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+		await admin.visitSiteEditor();
+
+		const contentAfter = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfter );
+	} );
+
+	test( 'should show the edited content after revert and clicking undo in the header toolbar', async ( {
+		page,
+		templateRevertUtils,
+	} ) => {
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		// Revert template and check state.
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+		const contentAfterSave = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).not.toEqual( contentAfterSave );
+
+		// Undo revert and check state again.
+		await page.click( 'role=button >> text="Undo"' );
```

Comment:
Why not just use the `name` attribute?

```suggestion
		await page.click( 'role=button[name="Undo"i]' );
```

Reply:
So this is interesting: turns out with the `>> text="Undo"` selector I was actually targeting the snackbar button 😅 

I tried changing to some more specific selectors:
```
const headerToolbar = page.locator( 'role=region[name="Header"]' );
await headerToolbar.click( 'role=button[name="Undo"]' );
```
but now the test is failing, apparently because it can't locate the button (debug mode shows it trying to click the centre section of the header, instead of the undo button). 

but if I change it to
```
await page.click( '.edit-site-header__toolbar [aria-label="Undo"]' );
```
it passes 😭  Any idea what might be going wrong here?

Reply:
Update: it also works with 
```
await page.click( 'role=region[name="Header"] >> role=button[name="Undo"]' );
```
so I'm going with that for now. I would have thought it would do exactly the same as the two-liner above that _didn't_ work though 😕

Reply:
I believe that's because [`locator.click()`](https://playwright.dev/docs/api/class-locator#locator-click) doesn't accept selectors as a parameter 😆